qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -149,60 +227,99 @@ def settings(self) -> Dict[str, Any]:
             "callback": self.callback,
         }
 
-    def minimize(
+    def ask(self) -> AskObject:
+        """
+        For gradient descent this method simply returns an
+        :class:`qiskit.algorithms.optimizers.AskObject` containing the current point for the
```

Comment:
This should be documented as a Returns. The type should get picked up from the typehint. Though really, since this implements the ask interface of the parent steppable optimizer, we normally would not have any docstring here letting Sphinx use the docstring from the parent. Same for tell below.

Reply:
So I should get rid of the documentation for these methods?

Reply:
Unless the docs here need to be different, to perhaps point out something for this sub-class, the function should conform to the docs listed in the parent. Unless it needs to be duplicated/enhanced for some reason its more usual to let Sphinx use the docstring from the parent method this overrides.

Reply:
Okay, I see. In this case maybe doing some documentation that explains how to use the data inside of AskData for this particular optimizer might be useful. What do you think? Should we explain it here or somewhere else?